store/src/test/java/com/dropbox/android/external/store4/impl/TransformingFetcherTest.kt

```diff
@@ -0,0 +1,175 @@
+package com.dropbox.android.external.store4.impl
+
+import com.dropbox.android.external.store4.FetcherResult
+import com.dropbox.android.external.store4.ResponseOrigin
+import com.dropbox.android.external.store4.StoreBuilder
+import com.dropbox.android.external.store4.StoreRequest
+import com.dropbox.android.external.store4.StoreResponse
+import com.dropbox.android.external.store4.TransformingFetcher
+import com.dropbox.android.external.store4.nonFlowValueFetcher
+import com.dropbox.android.external.store4.testutil.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+@FlowPreview
+class TransformingFetcherTest {
+
+    private val testScope = TestCoroutineScope()
+
+    @Test
+    fun `GIVEN transformer WHEN raw value THEN unwrapped value returned AND value is cached`() =
+        testScope.runBlockingTest {
+            val fetcher = TransformingFetcher<Int, Int, String>(
```

Comment:
since this is not public API anymore, maybe remove these tests or just inline it to do the transformation itself rather than relying on another class to do it ?